### PR TITLE
Add model

### PR DESCRIPTION
**Pull request by wakwak3125**
## WHAT
refs: https://github.com/creasty/torinos/issues/11
- [x] Create necessary models
- [x] Create migration files

### CODE REVIEW

**Comment by wakwak3125** [comment]
とりあえず、全部のマイグレーションが通るようにはなった。
@creasty 先生、レビューをお願いします。

File: data/migrations/20170702014400_create_projects.down.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+drop table notifications;
```
Comment (creasty) [inline data/migrations/20170702014400_create_projects.down.sql]: `projects`?

File: data/migrations/20170702014400_create_projects.up.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
+create table projects (
+    uuid varchar(255) primary key,
```
Comment (creasty) [inline data/migrations/20170702014400_create_projects.up.sql]: Postgres has precisely `uuid` type

File: data/migrations/20170702014400_create_projects.up.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    user_id int references users(id),
```
Comment (creasty) [inline data/migrations/20170702014400_create_projects.up.sql]: We'd like to delete projects that belong to a certain user if the user is deleted.
And by specifying `on delete cascade`, it'll be achieved.

File: data/migrations/20170702014400_create_projects.up.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    cart_file_content text,
```
Comment (creasty) [inline data/migrations/20170702014400_create_projects.up.sql]: Rename to `cartfile_content`, `podfile_lock_content`, and `pbxproj_content`, respectively.

c.f., https://github.com/creasty/torinos/issues?utf8=%E2%9C%93&q=label%3Aanalyze%20

File: data/migrations/20170702015658_create_notifications.up.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+create table notifications (
+    id serial primary key,
+    user_id int references users(id),
+    project_uuid varchar(255) not null references projects(uuid),
+    email varchar(255) not null,
```
Comment (creasty) [inline data/migrations/20170702015658_create_notifications.up.sql]: Add a compound index on `(project_uuid, email)`.

A link we're gonna send in mail will be like `https://torinos.io/projects/:uuid?email=:email`, which uses the index.

File: type/model/dependency.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package model
+
+// SourceCD
```
Comment (creasty) [inline type/model/dependency.go:3]: Enums can be represented as an aliased type.
Refer https://github.com/wantedly/yashima-news-timeline/blob/master/type/model/article_body_type.go

Comment (creasty) [reply type/model/dependency.go:3]: Fine with leaving it in the PR and fix it later in another, tho

**Review by creasty — COMMENTED** [review]
Besides some small points, looks basically okay 👍 

Btw, you've forgotten to follow this coding style.
https://github.com/wantedly/notifications/issues/16

File: type/model/dependency.go (hunk 6 above)
Comment (wakwak3125) [reply type/model/dependency.go:3]: Okay. Thanks 👍

**Comment by wakwak3125** [comment]
Thank you for reviewing. I applied your findings.
Could you review one more time?

File: type/model/project.go [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+// 1 = fetching
+// 2 = analyzing
+// 3 = done
+// 4 = error
+
+// Project has uploaded repository information.
+type Project struct {
+	UUID string `json:"uuid"`
+
+	CartFileContent       string `json:"cart_file_content"`
```
Comment (creasty) [inline type/model/project.go]: Could you make these properties consistent with the column names?

```
cartfile_content        text,
podfile_lock_content    text,
pbxproj_content         text,
```

**Review by creasty — COMMENTED** [review]
Perhaps, we will need to use https://github.com/guregu/null for nullable properties...

**Review by creasty — APPROVED** [review]
Looks good!

**Pull request merged**